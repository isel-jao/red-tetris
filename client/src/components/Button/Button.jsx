import React from 'react';
import { clsx } from 'clsx';

/**
 * Button component for use throughout the Red Tetris application
 * 
 * @param {Object} props - Component props
 * @param {string} [props.variant='primary'] - Button variant (primary, secondary, outline)
 * @param {string} [props.size='md'] - Button size (sm, md, lg)
 * @param {React.ReactNode} props.children - Button content
 * @param {string} [props.className] - Additional CSS classes
 * @param {React.ButtonHTMLAttributes} props.rest - Additional button attributes
 */
const Button = ({ 
  variant = 'primary', 
  size = 'md', 
  children, 
  className, 
  ...rest 
}) => {
  const baseClasses = 'rounded font-medium transition-colors focus:outline-none focus:ring-2';
  
  const variantClasses = {
    primary: 'bg-blue-600 text-white hover:bg-blue-700 focus:ring-blue-500',
    secondary: 'bg-gray-200 text-gray-900 hover:bg-gray-300 focus:ring-gray-500',
    outline: 'bg-transparent border border-blue-600 text-blue-600 hover:bg-blue-50 focus:ring-blue-500',
  };
  
  const sizeClasses = {
    sm: 'py-1 px-3 text-sm',
    md: 'py-2 px-4 text-base',
    lg: 'py-3 px-6 text-lg',
  };
  
  const buttonClasses = clsx(
    baseClasses,
    variantClasses[variant],
    sizeClasses[size],
    className
  );
  
  return (
    <button className={buttonClasses} {...rest}>
      {children}
    </button>
  );
};

export default Button;
