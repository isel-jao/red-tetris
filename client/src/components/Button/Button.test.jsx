import React from 'react';
import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import TestRenderer from 'react-test-renderer';
import Button from './Button';

describe('Button Component', () => {
  // Test using react-test-renderer for snapshot testing
  it('renders correctly with default props', () => {
    const tree = TestRenderer.create(<Button>Click me</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // Test using React Testing Library for interaction
  it('renders with correct text', () => {
    render(<Button>Click me</Button>);
    expect(screen.getByText('Click me')).toBeDefined();
  });

  // Test using props
  it('applies custom className when provided', () => {
    const { container } = render(<Button className="custom-btn">Custom Button</Button>);
    expect(container.firstChild).toHaveClass('custom-btn');
  });
});
