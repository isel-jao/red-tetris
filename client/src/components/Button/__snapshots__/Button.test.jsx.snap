// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button Component > renders correctly with default props 1`] = `null`;
